### PR TITLE
Paypal oneoff

### PR DESCRIPTION
**Pull request by octopusinvitro**
This PR implements Paypal one-off payments in the frontend

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,9 +4,10 @@ source 'https://rubygems.org'
 ruby '2.5.0'
 
 gem 'donation_system', git: 'https://github.com/survival/donation-system',
-                       tag: 'v1.0.0'
+                       branch: 'paypal-one-off-salesforce'
```
Comment (samstickland) [inline Gemfile]: I presume this will be changed back before merging?

File: package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "scripts": {
     "sass": "node-sass public/scss/main.scss public/css/main.css --output-style compressed",
     "linter": "eslint public/js/src/*js public/js/spec/*js",
-    "uglify": "uglifyjs public/js/src/ui.js public/js/src/stripe_wrapper.js -c -m -o public/js/bundle.js",
+    "uglify": "uglifyjs public/js/src/ui.js public/js/src/stripe_wrapper.js public/js/src/paypal_wrapper.js -c -m -o public/js/bundle.js",
```
Comment (samstickland) [inline package.json:8]: Any reason we don't uglify everything? Like `linter`?

File: spec/donations_spec.rb [hunk 3]
```diff
@@ -4,37 +4,79 @@
...
+      described_class.donate('param' => 'irrelevant')
       expect(Helpers::InputSanitizer).to have_received(:execute)
-        .with('foo' => 'bar')
+        .with('param' => 'irrelevant')
     end
 
     it 'attempts a donation' do
       data = Helpers::DonationData.new(
-        '', '', '', false, '', '', '', '', '', '', '', ''
+        '', '', '', false, '', '', '', '', '', '', '', '', ''
```
Comment (samstickland) [inline spec/donations_spec.rb]: These structs are getting a little hard to read. What do you think about using Ruby 2.5's new 'keyword_init: true` ?

**Review by samstickland — APPROVED** [review]

File: Gemfile (hunk 1 above)
Comment (octopusinvitro) [reply Gemfile]: yes

File: package.json (hunk 2 above)
Comment (octopusinvitro) [reply package.json:8]: Ah, I understand your comment now. The order in which the js files are concatenated is important

File: spec/donations_spec.rb (hunk 3 above)
Comment (octopusinvitro) [reply spec/donations_spec.rb]: Cool stuff! 👍

**Pull request merged**